drivers/video/video_common.c

```diff
@@ -83,3 +84,45 @@ void video_buffer_release(struct video_buffer *vbuf)
 		VIDEO_COMMON_FREE(block->data);
 	}
 }
+
+int video_get_range_u32(unsigned int cid, uint32_t *value, uint32_t min, uint32_t max, uint32_t def)
```

Comment:
Why do we only check uint32_t ? What if the control values are of other types, e.g. uint64_t (pixel_rate, link frequencies), menu (test pattern), etc. ?

Reply:
it cannot be `uint64_t` (as it does not fit in a pointer in some architectures supported by Zephyr) but it can be `uint64_t *`.

Maybe a first compromise is to support `int32_t` and `uint32_t` (or to be more correct, `uintptr_t` and `intptr_t`) is good enough for a first cycle? I am updating this branch to illustrate this.

Reply:
I am waiting before introducing `bool`, `menu`, `bitmap` and [extra types](https://git.kernel.org//pub/scm/linux/kernel/git/torvalds/linux.git/tree/include/uapi/linux/videodev2.h#n1909) as I tried to focus on the mechanism that enable fetching a range that would fit the current API.

Let me know if I should include them on this PR, I'd be glad to give it a try!